source/code/plugin/health/health_monitor_utils.rb

```diff
@@ -0,0 +1,369 @@
+                            cpu_capacity_json = KubernetesApiClient.parseNodeLimits(node_inventory, "capacity", "cpu", "cpuCapacityNanoCores")
+                            if !cpu_capacity_json.nil?
+                                cpu_capacity_json.each do |cpu_capacity_node|
+                                    if !cpu_capacity_node['DataItems'][0]['Collections'][0]['Value'].to_s.nil?
+                                        cluster_cpu_capacity += cpu_capacity_node['DataItems'][0]['Collections'][0]['Value']
+                                    end
+                                end
+                                log.info "Cluster CPU Limit #{cluster_cpu_capacity}"
+                            else
+                                log.info "Error getting cpu_capacity"
+                            end
+                            memory_capacity_json = KubernetesApiClient.parseNodeLimits(node_inventory, "capacity", "memory", "memoryCapacityBytes")
+                            if !memory_capacity_json.nil?
+                                memory_capacity_json.each do |memory_capacity_node|
+                                    if !memory_capacity_node['DataItems'][0]['Collections'][0]['Value'].to_s.nil?
+                                        cluster_memory_capacity += memory_capacity_node['DataItems'][0]['Collections'][0]['Value']
+                                    end
+                                end
+                                log.info "Cluster Memory Limit #{cluster_memory_capacity}"
+                            else
+                                log.info "Error getting memory_capacity"
+                            end
+                        end
+                    else
+                        log.info "Unable to get cpu and memory capacity"
+                        return [0.0, 0.0]
+                    end
+                    return [cluster_cpu_capacity, cluster_memory_capacity]
+                rescue => e
+                    log.info e
+                end
+            end
+
+            def refresh_kubernetes_api_data(log, hostName, force: false)
+                #log.debug "refresh_kubernetes_api_data"
+                if ( ((Time.now.utc - Time.parse(@@last_refresh_time)) / 60 ) < 5.0 && !force)
+                    log.debug "Less than 5 minutes since last refresh at #{@@last_refresh_time}"
+                    return
+                end
+                if force
+                    log.debug "Force Refresh"
+                end
+
+                begin
+                    @@nodeInventory = JSON.parse(KubernetesApiClient.getKubeResourceInfo("nodes").body)
+                    if !hostName.nil?
+                        podInventory = JSON.parse(KubernetesApiClient.getKubeResourceInfo("pods?fieldSelector=spec.nodeName%3D#{hostName}").body)
+                    else
+                        podInventory = JSON.parse(KubernetesApiClient.getKubeResourceInfo("pods").body)
```

Comment:
This will return a huge number of pods on a huge cluster. We already have this issue in the rs, and it might just double. 100 nodes with ~150 pods in each node (Marko's cluster) will have 15k pods with more than 100MB payload parsing. Can you please test this in bigger clusters (mainly AKS-engine)

Reply:
Agreed. I am planning to use the limit and continue params in the API in a future release. Was planning on doing that before public preview.